### DISCUSSION TITLE
How do I know ContentMenu's Parent Control?

### DISCUSSION THREAD

**Discussion by xinghaiyege** [AvaloniaUI/Avalonia#10619] [Q&A] [answered]
The code in WPF is like this, but using the Avalonia UI, I don't know how to get Parent Control.

```C#
TextBlock textblock= ContextMenuService.GetPlacementTarget(LogicalTreeHelper.GetParent(sender as MenuItem)) as TextBlock;
string data = textblock.Text;
```

**Comment by robloo** [comment, marked as answer]
There are extension and helper methods to navigate the logical or visual tree:

https://reference.avaloniaui.net/api/Avalonia.LogicalTree/LogicalExtensions/

https://reference.avaloniaui.net/api/Avalonia.VisualTree/VisualExtensions/

Edit: Added logical tree extensions which is what is needed in question above.
